autogpt_platform/frontend/src/hooks/useAgentGraph.tsx

```diff
@@ -461,6 +464,12 @@ export default function useAgentGraph(
         flowID,
         flowExecutionID,
       );
+
+      // Set graph execution error from the initial fetch
+      if (execution.status === "FAILED" && execution.stats?.error) {
+        setGraphExecutionError(execution.stats.error);
+      }
```

Comment:
In the edge case, we don't have a `stats?. error` returned, I would still set a generic error so it shows up in the UI?

Reply:
Good idea, updated